### PR TITLE
Add Mediator pattern

### PR DESCRIPTION
**Pull request by anicholson**

### CODE REVIEW

**Comment by anicholson** [comment]
Part of #9 :)

**Comment by veelenga** [comment]
Please rebase, so we will be sure travis passes.

**Comment by anicholson** [comment]
@veelenga done :)

File: behavioral/mediator.cr [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  end
+
+  def perform_move(move, fighter)
+    if can_perform_move?(move, fighter)
+      if (fighter == player1)
+        player2.as(Fighter).receive(move, fighter)
+      elsif (fighter == player2)
+        player1.as(Fighter).receive(move, fighter)
+      else
+        # not a player in this match
```
Comment (veelenga) [inline behavioral/mediator.cr]: Do we need this empty else?

**Comment by veelenga** [comment]
Looks good except one minor comment. Thanks

**Comment by anicholson** [comment]
@veelenga fixed and added outputs to the bottom for extra grokability.

**Comment by veelenga** [comment]
Thx

**Pull request merged**